src/EventListener/StorageListener.php

```diff
@@ -99,7 +99,11 @@ public function preHydrate(HydrationEvent $event)
 				foreach ($localeData as $key => $value) {
 					if (isset($contentType['fields'][$key]['type']) && $contentType['fields'][$key]['type'] !== 'repeater') {
 						$subject[$key] = is_array($value) ? json_encode($value) : $value;
-					}
+					}  elseif ( $key == 'templatefields') {
+						foreach ($value as $i => $field) {
+							$templatefields[$i]= is_array($field) ? json_encode($field) : $field;
```

Comment:
I see that there have snuck in tabs here and in a few other places (and I saw that there were a bunch of places they had snuck into the master repo too) could you replace them with 4 spaces and I'll clean up the ones in the core repo post-merge?